cpp/src/utilities/high_res_timer.hpp

```diff
@@ -85,6 +85,17 @@ class HighResTimer {
     timers.clear();
   }
 
+  double average_runtime(std::string const &label)
+  {
+    auto it = timers.find(label);
+    if (it != timers.end()) {
+      return (static_cast<double>(it->second.second) / (1000000.0 * it->second.first));
+    } else {
+      std::cerr << "ERROR: timing label: " << label << "not found.";
```

Comment:
"ERROR" -> "WARNING"

Reply:
Why warning? If the user asks for a timing result that doesn't exist then something is erroneous; i.e., not something that can be ignored, like a warning is.

The debate might be whether to `throw` rather than dump an error message to `std::cerr`. But that would require surrounding this call by a `try`-block in tests. Which might not be a bad idea, actually.

Reply:
For all intends and purposes, not finding a label is a failed test. Hence, I've added exception handling for this as a mechanism to communicate back to the tester that it should consider the test failed on label not found.

Reply:
Thanks.  FYI, I should have also mentioned this before, but the other issue with printing `ERROR` is that some scripts and CI systems look for that string in the output and may interpret it in a way we don't intend them to. As a best practice I usually throw/raise to communicate that I cannot reliably continue, and I use warnings for all other unclear situations. This looked like a warning since you were allowing the test to continue, but I like throwing better since like you said, it's an erroneous condition.